pmm-app/tests/pages/remoteInstancesPage.js

```diff
@@ -215,4 +230,33 @@ module.exports = {
     I.fillField(this.fields.replicationSet, 'rds56-replication');
     I.scrollPageToBottom();
   },
+
+  parseURL(url) {
+    I.waitForVisible(this.fields.urlInput, 30);
+    I.fillField(this.fields.urlInput, url);
+    I.click(this.fields.parseUrlButton);
+  },
+
+  async checkParsing(metricsPath, credentials) {
+    const grabbedHostname = await I.grabValueFrom(this.fields.hostName);
+    const grabbedMetricPath = await I.grabValueFrom(this.fields.metricsPath);
+    const grabbedPort = await I.grabValueFrom(this.fields.portNumber);
+    const grabbedCredentials = await I.grabValueFrom(this.fields.userName);
+    const httpsLocator = '//label[contains(@class, "active") and contains(text(), "HTTPS")]';
+
+    assert.ok(grabbedHostname === process.env.MONITORING_HOST, `Hostname is not parsed correctly: ${grabbedHostname}`);
+    assert.ok(grabbedMetricPath === metricsPath, `Metrics path is not parsed correctly: ${grabbedMetricPath}`);
+    assert.ok(grabbedPort === process.env.EXTERNAL_EXPORTER_PORT, `Port is not parsed correctly: ${grabbedPort}`);
+    assert.ok(grabbedCredentials === credentials, `Username is not parsed correctly: ${grabbedCredentials}`);
+    assert.ok(grabbedCredentials === credentials, `Password is not parsed correctly: ${grabbedCredentials}`);
+    I.waitForVisible(httpsLocator, 30);
+  },
+
+  checkRequiredField() {
+    const hostname = '//div[@data-qa="address-field-container"]/child::div[@data-qa="address-field-error-message"]';
```

Comment:
locate doesn't help here?

Reply:
That one seems to be too long, I guess we could as well grab only the `data-qa="address-field-error-message"` since it seems to be unique.